### ISSUE TITLE
bug: screen reader does not speak updated ion-select value

### ISSUE THREAD

**Issue by ehorodyski-ionic** [ionic-team/ionic-framework#22278]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**

When switching the value of an `ion-select` with screen reader technology enabled, the following is observed:

- **Android**: After updating the value, the new value is not read back by the screen reader. Instead, "Button" is read back.
- **iOS**: After updating the value, the new value is read back by the screen reader, along with several utterances such as "Button, List End, End, End, Main"

**Expected behavior:**

- **Android**: The updated value should be read back by the screen reader

**Steps to reproduce:**

1. Clone the following reproduction: https://github.com/ehorodyski-ionic/fw-a11y-ion-select
2. Run the application on Android with TalkBack enabled.
3. Change the selected item in the input field from "Brown" to "Red".
4. Observe upon dismissal of the selection action sheet, the updated value is not spoken back.

**Related code:**

https://github.com/ehorodyski-ionic/fw-a11y-ion-select

**Other information:**

Related Issues:

- https://github.com/ionic-team/ionic-framework/issues/21601
- https://github.com/ionic-team/ionic-framework/issues/21548 

**Ionic info:**

```
Ionic:

   Ionic CLI                     : 6.11.9 (/Users/eric/npm-global/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.3.4
   @angular-devkit/build-angular : 0.1000.8
   @angular-devkit/schematics    : 10.0.8
   @angular/cli                  : 10.0.8
   @ionic/angular-toolkit        : 2.3.3

Capacitor:

   Capacitor CLI   : 2.4.2
   @capacitor/core : 2.4.2

Utility:

   cordova-res (update available: 0.15.1) : 0.11.0
   native-run (update available: 1.2.1)   : 1.2.0

System:

   NodeJS : v12.16.1 (/usr/local/bin/node)
   npm    : 6.14.8
   OS     : macOS Catalina
```

**Comment by brandyscarney** [comment]
This is fixed by https://github.com/ionic-team/ionic-framework/commit/04b874e32a65588ca79eda9399ab7e9d86a3cb77 and https://github.com/ionic-team/ionic-framework/commit/ea52db66f05a185fed6b2e849734a7ffa1c6c6ea
